lib/iris/analysis/__init__.py

```diff
@@ -1419,36 +1419,45 @@ def interp_order(length):
 """
 
 
-def _build_dask_mdtol_function(dask_nanfunction):
+def _build_dask_mdtol_function(dask_stats_function):
     """
     Make a wrapped dask statistic function that supports the 'mdtol' keyword.
 
-    'dask_nanfunction' must be a statistical function that treats NaNs as
-    missing data, and which has the basic call signature
-    "dask_nanfunction(data, axis, **kwargs)".
+    'dask_function' must be a dask statistical function, compatible with the
+    call signature : "dask_stats_function(data, axis, **kwargs)".
+    It must be masked-data tolerant, i.e. it ignores masked input points and
+    performs a calculation on only the unmasked points.
+    For example, mean([1, --, 2]) = (1 + 2) / 2 = 1.5.
 
     The returned value is a new function operating on dask arrays.
-    It has the call signature "stat(data, axis=-1, mdtol=None, *kwargs)".
+    It has the call signature "stat(data, axis=-1, mdtol=None, **kwargs)".
 
     """
-    @wraps(dask_nanfunction)
+    @wraps(dask_stats_function)
     def inner_stat(array, axis=-1, mdtol=None, **kwargs):
-        dask_result = dask_nanfunction(array, axis=axis, **kwargs)
+        # Call the statistic to get the basic result (missing-data tolerant).
+        dask_result = dask_stats_function(array, axis=axis, **kwargs)
         if mdtol is None:
```

Comment:
It does always bother me to do equality testing on floats, though I suppose this is only a shortcut.

So maybe this should say `mdtol is None or mdtol >= 1.0`, 
making any mdtol > 1 equivalent to mdtol == 1.
That makes practical sense, meaning "valid if there is even one good point".
( As internally in the code, we have `boolean_mask = masked_point_fractions > mdtol` )

However, it's slightly inconsistent with the effect of mdtol < 0.0 :  
Whereas "mdtol=0" means "no missing points", "mdtol<0" will mask everything, regardless.

Reply:
> though I suppose this is only a shortcut

Yep.